### PR TITLE
Keycloak oidc client

### PR DESCRIPTION
**Pull request by julioa**
Basic keycloak oidc client.

### CODE REVIEW

**Comment by julioa** [comment]
Hi

I don't know if I'm bothering you with these PRs but I hope the code I'm writing for some prototypes I'm building may be useful for others.

**Comment by leleuj** [comment]
Interesting! Contributions (almost) never bother me ;-)

I'll review it.

File: pac4j-oidc/src/main/java/org/pac4j/oidc/client/KeycloakOidcClient.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        CommonHelper.assertNotNull("configuration", getConfiguration());
+        String baseUri = getConfiguration().getCustomParam("authUrl");
+        CommonHelper.assertNotNull("authUrl", baseUri);
+        String realm = getConfiguration().getCustomParam("realm");
+        CommonHelper.assertNotNull("realm", realm);
+        getConfiguration().setDiscoveryURI(baseUri+"/realms/"+realm+"/.well-known/openid-configuration");
+        final OidcProfileCreator<KeycloakOidcProfile> profileCreator = new OidcProfileCreator<>(getConfiguration());
+        profileCreator.setProfileDefinition(new OidcProfileDefinition<>(x -> new KeycloakOidcProfile()));
+        setProfileCreator(profileCreator);
+        //setLogoutActionBuilder(new GoogleLogoutActionBuilder<>());
```
Comment (leleuj) [inline pac4j-oidc/src/main/java/org/pac4j/oidc/client/KeycloakOidcClient.java]: Don't keep such commented line of code.

File: pac4j-oidc/src/main/java/org/pac4j/oidc/config/OidcConfiguration.java [hunk 2]
```diff
@@ -157,6 +157,14 @@ public void setScope(final String scope) {
         return customParams;
     }
 
+    public String getCustomParam(String name) {
+        if (customParams != null) {
```
Comment (leleuj) [inline pac4j-oidc/src/main/java/org/pac4j/oidc/config/OidcConfiguration.java]: This check is useless as the `customParams` can never be `null`.

**Comment by leleuj** [comment]
A few comments, but overall, it looks good.

Maybe it would be interesting to create some manual test for it, like we did for IdentityServer4 for example (https://github.com/pac4j/pac4j/blob/master/pac4j-oidc/src/test/java/org/pac4j/oidc/run/RunIdentityServer4.java).

File: pac4j-oidc/src/main/java/org/pac4j/oidc/client/KeycloakOidcClient.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ *   <li>authUrl : Base auth server url, e.g., https://keycloak.example.com/auth </li>
+ *   <li>realm : keycloak realm</li>
+ * </ul>
+ *
+ * @author Julio Arrebola
+ * @since 2.0.0
+ */
+public class KeycloakOidcClient extends OidcClient<KeycloakOidcProfile> {
+
+    private static final long serialVersionUID = -665205280702028258L;
```
Comment (leleuj) [inline pac4j-oidc/src/main/java/org/pac4j/oidc/client/KeycloakOidcClient.java]: The `serialVersionUID` should be in the profile class, not in the client one.

**Comment by leleuj** [comment]
Thanks. Merging

**Pull request merged**